src/rasta/c/rasta_new.c

```diff
@@ -595,17 +595,35 @@ void sr_retransmit_data(struct rasta_receive_handle *h, struct rasta_connection
     }
 
     // close retransmission with heartbeat
-    send_Heartbeat(h->mux,connection);
+    send_Heartbeat(h->mux,connection, 1);
 }
+
+struct timed_event_data {
+    void * handle;
+    int connection_index;
+};
+
+#define MAX_CONNECTIONS_SUPPORTED 4
```

Comment:
Possibly add documentation (comments): What is a connection? (a pair of two communicating parties, possibly over multiple channels?) 

Why is the limit 4? Why is there a limit at all (thinking about an interlocking which possibly has many connections to field elements)? Is this more restrictive than before the refactoring?